### ISSUE TITLE
Guard Gem is not compatible with Ruby-LSP

### ISSUE THREAD

**Issue by richjdsmith** [Shopify/ruby-lsp#1105]
### Ruby version

3.2.0

### Code snippet

Ruby-LSP fails to index files when the Guard Gem is installed and in the Gemfile. 

```
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '3.2.0'
...
...

group :development, :test do
  # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
  # gem 'awesome_print'
  # gem 'binding_of_caller'
  # gem 'guard-livereload', '~> 2.4', require: false
  # gem 'rack-livereload'
  # gem 'web-console'

  gem 'guard'
```

### Description

I updated, reinstalled, and tried past versions of Ruby-LSP. I have tested it and it appears not to work inside or outside VS Code. The presence of the Guard gem seems to break Ruby-LSP's ability to index any files. 

This is the output:

```
Aborted
[Error - 12:13:55 PM] Server process exited with code 134.
[Info  - 12:13:55 PM] Connection to server got closed. Server will restart.
true
[Error - 12:13:55 PM] Request textDocument/diagnostic failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at Ou.pullAsync (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:6413)
[Error - 12:13:55 PM] Request textDocument/diagnostic failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at Ou.pullAsync (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:6413)
[Error - 12:13:55 PM] Request textDocument/documentSymbol failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at u (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:55784)
    at r.provideDocumentSymbols (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:37581)
[Error - 12:13:55 PM] Request textDocument/codeAction failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at j.provideCodeActions (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:42299)
[Error - 12:13:55 PM] Request textDocument/documentLink failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at me.provideLinks (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:57867)
[Error - 12:13:55 PM] Request textDocument/inlayHint failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at d (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:105196)
    at H.provideInlayHints (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:56138)
[Error - 12:13:55 PM] Request textDocument/semanticTokens/range failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at x.provideDocumentRangeSemanticTokens (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:50128)
[Error - 12:13:55 PM] Request textDocument/foldingRange failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at ne.provideFoldingRanges (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:59245)
[Error - 12:13:55 PM] Request textDocument/codeLens failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at Object.provideCodeLenses (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:44:59246)
    at i.provideCodeLenses (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:38624)
[Error - 12:13:55 PM] Request textDocument/semanticTokens/full failed.
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at V.provideDocumentSemanticTokens (/home/richsmith/.vscode-server/bin/f1b07bd25dfad64b0167beb15359ae573aecd2cc/out/vs/workbench/api/node/extensionHostProcess.js:128:48652)
[Error - 12:13:55 PM] Document pull failed for text document file:///home/richsmith/workspace/tekamarsite/app/models/calculators/payment_calculator.rb
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at Ou.pullAsync (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:6413)
[Error - 12:13:55 PM] Document pull failed for text document file:///home/richsmith/workspace/tekamarsite/Gemfile
TypeError: Cannot read properties of undefined (reading 'errorClass')
    at us.benchmarkMiddleware (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:50:3654)
    at Ou.pullAsync (/home/richsmith/.vscode-server/extensions/shopify.ruby-lsp-0.4.10/out/extension.js:39:6413)
Ruby LSP> Skipping custom bundle setup since /home/richsmith/workspace/tekamarsite/.ruby-lsp/Gemfile.lock already exists and is up to date
Ruby LSP> Running bundle install for the custom bundle. This may take a while...
The Gemfile's dependencies are satisfied
```

### Expected output

I would expect (or maybe I should not?) Ruby-LSP to work with Guard. That said, this error report is also for anyone looking to troubleshoot, as I cannot find anything else on it, and it took me removing all gems and slowly adding them back to isolate the issue.

**Comment by andyw8** [comment]
Guard has a dependency on Lumberjack so this likely the same issue as https://github.com/Shopify/ruby-lsp/issues/1063

**Comment by andyw8** [comment]
I discovered a secondary issue:
https://github.com/Shopify/ruby-lsp/issues/1108

Since these are now both documented, I'll close this issue as a duplicate.

**Comment by andyw8** [comment]
(thank you for the detailed bug report @richjdsmith).

**Comment by steveclarke** [comment]
@richjdsmith This is exactly the problem that I had. I discovered further that the problem lies in the Lumberjack gem that is a dependency of Guard. So I could keep using Guard, I forked Lumberjack and removed the offending magic comments. Feel free to use my fork for the interim (https://github.com/steveclarke/lumberjack) although I understand an update to Ruby LSP is imminent that will fix the problem.
